syntax = 'proto3';

service Products {
  rpc GetAll (Empty) returns (ProductsResponseList) {}
  rpc Get (ProductsRequestId) returns (ProductsResponse) {}
  rpc Insert (ProductsInsertRequest) returns (ProductsResponse) {}
  rpc Update (ProductsUpdateRequest) returns (ProductsResponse) {}
  rpc Remove (ProductsRequestId) returns (Empty) {}
}

message Empty {}

message ProductsInsertRequest {
  string name = 1;
  float price = 2;
}

message ProductsUpdateRequest {
  string id = 1;
  string name = 2;
  float price = 3;
}

message ProductsRequestId {
  string id = 1;
}

message ProductsResponse {
  string id = 1;
  string name = 2;
  float price = 3;
}

message ProductsResponseList {
  repeated ProductsResponse productsResponseList = 1;
}